Lib/urllib/request.py

```diff
@@ -1182,6 +1182,8 @@ def get_algorithm_impls(self, algorithm):
         elif algorithm == 'SHA':
             H = lambda x: hashlib.sha1(x.encode("ascii")).hexdigest()
         # XXX MD5-sess
+        elif algorithm == 'SHA-256':
```

Comment:
can you move the `XXX MD5-sess` before the else and not before the `elif`? thanks. In addition, mention that the algorithm names are taken from https://datatracker.ietf.org/doc/html/rfc7616#section-6.1.

Reply:
done:

- https://github.com/python/cpython/pull/128193/commits/5c214022d6aff41069b65a587cd2632777e158cd
- https://github.com/python/cpython/pull/128193/commits/aea1b4ec777ddde9cc8ecfdc99e4f3762313c08d